feat(n8n Form Trigger Node): Form Improvements

dana-gill opened the pull request:
## Summary

- Allows users to use HTML in Form descriptions and also sanitizes the HTML to prevent Javascript from being executed.
For this PR, we are only allowing a select number of HTML elements. We can easily add more by adding more to the `sanitize` [options](https://www.npmjs.com/package/sanitize-html#default-options).
- We are also populating the rendered Form's description metadata with the form description
- Add OG metadata to improve sharing across the Internet

Node for testing:

```
{
  "nodes": [
    {
      "parameters": {
        "formTitle": "form",
        "formDescription": "<script>alert('hello world')</script>\n<i>hello</i>",
        "formFields": {
          "values": [
            {
              "fieldLabel": "name"
            },
            {
              "fieldLabel": "age",
              "fieldType": "number",
              "requiredField": true
            }
          ]
        },
        "options": {}
      },
      "type": "n8n-nodes-base.formTrigger",
      "typeVersion": 2.2,
      "position": [
        -100,
        -60
      ],
      "id": "51631ae1-11ac-4289-942d-159e90013470",
      "name": "On form submission",
      "webhookId": "eb1eed7e-5b3e-4582-962f-fd955d9f44f1"
    }
  ],
  "connections": {
    "On form submission": {
      "main": [
        []
      ]
    }
  },
  "pinData": {}
}
```

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/NODE-2227/sanitize-rendered-html-for-form-descriptions
https://linear.app/n8n/issue/NODE-2228/render-html-for-form-descriptions
https://linear.app/n8n/issue/NODE-2226/description-parameter-html-formatting
https://linear.app/n8n/issue/NODE-2237/form-title-and-form-description-metadata

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) <!--
   **Remember, the title automatically goes into the changelog.
   Use `(no-changelog)` otherwise.**
-->
- [x] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included. <!--
   A bug is not considered fixed, unless a test is added to prevent it from happening again.
   A feature is not complete without tests.
-->
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

Commit:
feat(Form Node): Enable HTML in form description

Commit:
Add Sanitize HTML

Commit:
Test sanitize-html

Commit:
Generalize executeFunctions for Form util tests

michael-radency reviewed (commented):
LGTM, some suggestions:
- update parameter description to mention that field accepts HTML
- add  'target' and 'rel' to `allowedAttributes`
- extend `allowedTags` by 'u', 'sub', 'sup', 'code', 'pre', 'span', 'br'

is there a way to tell if not allowed attributes/tags used? if so worth adding post execution hint

Commit:
PR requests and move sanitize HTML to separate function

michael-radency commented:
> > is there a way to tell if not allowed attributes/tags used? if so worth adding post execution hint
> 
> I think we would be able to do this by doing something like `context.getNodeParameter('formDescription', '') === sanitize(context.getNodeParameter('formDescription', ''))` so this is possible.
> 
> Do you have an idea of the best way to add the post execution hint? From my understanding, `formWebhook` is called when the Form is rendered.

never mind, I just tested - `NodeExecutionOutput` hints does not work with trigger and we do not have helper to sanitize HTML in expressions, so we could not define `displayCondition` in node's `hints` property either

michael-radency reviewed (dismissed):
LGTM :tada:

Commit:
Use form description for metadata of form

Commit:
Test parser for form description

michael-radency commented on packages/nodes-base/nodes/Form/utils.ts:
```diff
@@ -16,13 +16,49 @@ import {
+	});
+}
+
+export function createDescriptionMetadata(description: string) {
+	return description === ''
+		? 'n8n form'
+		: description.replace(/^\s*\n+|<\/?[^>]+(>|$)/g, '').slice(0, 150);
+}
```
@dana-gill would be always `false`

michael-radency replied:
sorry, mb, for some reason read it as `'' ? ... : ...`

michael-radency reviewed (commented):
should we have separate parameter in options for form description metatada with fallback to description?

dana-gill commented:
> should we have separate parameter in options for form description metatada with fallback to description?

I thought about doing that especially during [this Slack conversation](https://n8nio.slack.com/archives/C06K5TQRJJD/p1736854500257469). I get the feeling that we would want to do this as part of an initiative to allow users to set metadata themselves. I also added it under Various Requests in [this](https://www.notion.so/n8n/n8n-Form-and-FormTrigger-385708f93d8b4c698cb7573b4ff8b997) doc. As a result, I thought of taking a leaner and less invasive approach since having to add an additional field would require touching many parts of the Form node -- at least 3-4 functions plus the `*.d.ts` files. Let me know if you think this makes sense :)

michael-radency reviewed (dismissed):
LGTM

Commit:
Add Open Graph metadata

We cannot use relative paths for the image [1] so we are using the logo
in GitHub

[1]: https://stackoverflow.com/questions/9858577/open-graph-can-resolve-relative-url

Commit:
Incorporate design review feedback

janober commented:
Got released with [`n8n@1.76.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.76.0)
